src/sentry/plugins/providers/integration_repository.py

```diff
@@ -130,29 +139,40 @@ def create_repository(
                 setattr(repo, field_name, field_value)
             # also update the status if it was in a bad state
             repo.status = ObjectStatus.ACTIVE
-            repo.save()
+            repository_service.update_repository(organization_id=organization.id, update=repo)
         else:
+            create_repository = RpcCreateRepository.parse_obj(
+                {**repo_update_params, "status": ObjectStatus.ACTIVE}
+            )
+            new_repository = repository_service.create_repository(
+                organization_id=organization.id, create=create_repository
+            )
+            if new_repository is not None:
+                return result, new_repository
+
+            # Try to delete webhook we just created
             try:
-                with transaction.atomic(router.db_for_write(Repository)):
-                    repo = Repository.objects.create(
-                        organization_id=organization.id, **repo_update_params
-                    )
-            except IntegrityError:
-                # Try to delete webhook we just created
-                try:
-                    repo = Repository(organization_id=organization.id, **repo_update_params)
-                    self.on_delete_repository(repo)
-                except IntegrationError:
-                    pass
-
-                # if possible update the repo with matching integration
-                repo = Repository.objects.filter(
+                repo = Repository(organization_id=organization.id, **repo_update_params)
+                self.on_delete_repository(repo)
+            except IntegrationError:
+                pass
+
+            # if possible update the repo with matching integration
+            repositories = repository_service.get_repositories(
+                organization_id=organization.id,
+                integration_id=integration_id,
+                external_id=external_id,
+            )
+            repo = repositories[0] if repositories else None
+            if repo:
```

Comment:
there shouldn't be more than 1 repo that we update, but the original code updates all of them if there happens to be duplicates. should that be accounted for here?